scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java

```diff
@@ -993,7 +988,8 @@ private void dispatchSchedulerStateUpdated(SchedulerEvent eventType) {
             if (logger.isDebugEnabled()) {
                 logger.debug("event [" + eventType.toString() + "]");
             }
-            for (ListeningUser userId : identifications.values()) {
+            for (UserAndCredentials userAndCredentials : identifications.values()) {
+                ListeningUser userId = userAndCredentials.getListeningUser();
```

Comment:
Maybe put the declaration of userId before the loop

Reply:
To create a single ListeningUser object and initialize it each iteration.

Reply:
it is not creation, it is declaration, which changes nothing if outside the loop. Each element of identifications contains a different user, so there is no interest in simply putting the declaration of userId outside the loop.